esp/services/ws_cloud/ws_cloudService.cpp

```diff
@@ -59,6 +71,60 @@ bool CWsCloudEx::onGetPODs(IEspContext& context, IEspGetPODsRequest& req, IEspGe
     return true;
 }
 
+void addPublicService(IPropertyTree& serviceTree, StringBuffer& publicServices)
+{
+    delimitJSON(publicServices);
+
+    //Remove managedFields from the serviceTree.
+    IPropertyTree* metadataTree = serviceTree.queryPropTree(svcMetadata);
+    if (metadataTree)
+    {
+        ICopyArrayOf<IPropertyTree> managedFieldsTrees;
+        Owned<IPropertyTreeIterator> managedFieldsItr = metadataTree->getElements(svcMetadataManagedFields);
+        ForEach(*managedFieldsItr)
+            managedFieldsTrees.append(managedFieldsItr->query());
+        ForEachItemIn(i, managedFieldsTrees)
+            metadataTree->removeTree(&managedFieldsTrees.item(i));
+    }
+
+    toJSON(&serviceTree, publicServices);
+}
+
+//The code below reads 'public' services. A 'public' service is a service if the EXTERNAL-IP returned by
+// 'kubectl get svc' for the service is not <none> or <unknown>. The code is based on the getServiceExternalIP() in
+// https://github.com/kubernetes/kubernetes/blob/759785ea147bc13945d521eaba4a6592cbc0675f/pkg/printers/internalversion/printers.go#L1125.
+// The getServiceExternalIP() returns <unknown> if the service type is not "LoadBalancer", "ExternalName", "ClusterIP" or "NodePort".
+// When the service type is "ClusterIP" or "NodePort", the getServiceExternalIP() returns <none> if the "externalIPs" is not defined.
+const char* CWsCloudEx::readPublicServices(const char* allServices, StringBuffer& publicServices)
```

Comment:
Trivial, but it will be much much more intuitive to remember what these functions do if the names were more about "building json" rather than  "read" or add.

Perhaps buildJsonPublicServices() and above "addJsonPublicService()"?

Reply:
I will change them and move the addJsonPublicService() to the CWsCloudEx class.